### PR TITLE
[Sql Server] (No.4) add sql server delete logic.

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServer.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
     SqlServerEntity entity();
 
-    void delete();
+    ISqlServerDeleter<? extends ISqlServer> delete();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServer.java:16]: void delete(resource group name, name)

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServer.java:16]: ISqlServerDeleter support both **delete by ID** and delete by Resource Group & name.

Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServer.java:16]: You may get `ISqlServer` by both `id` and `rg&name`, so a simple delete method is enough
```java
Azure.az(AzureSqlServer.class).sqlServer("xxx-xx-xxx").delete()
Azure.az(AzureSqlServer.class).sqlServer("rg-xxx", "name-xxx").delete()
```

File: azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java [hunk 2]
```diff
@@ -50,11 +49,12 @@ public SqlServerEntity entity() {
...
-        return new SqlServerDeleter()
-            .withId(entity.getId())
-            .withName(entity.getName())
-            .withResourceGroup(entity.getResourceGroup());
+    public void delete() {
+        if (StringUtils.isNotBlank(entity.getId())) {
+            SqlServer.this.manager.sqlServers().deleteById(entity.getId());
+        } else if (StringUtils.isNotBlank(entity.getResourceGroup()) && StringUtils.isNotBlank(entity.getName())) {
+            SqlServer.this.manager.sqlServers().deleteByResourceGroup(entity.getResourceGroup(), entity.getName());
+        }
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java:57]: throw exception when the delete is not executed

**Review by andxu — APPROVED** [review]

**Pull request merged**